apps/desktop/src/autofill/main/main-desktop-autotype.service.ts

```diff
@@ -1,33 +1,73 @@
+import { ipcMain, globalShortcut } from "electron";
+
 import { autotype } from "@bitwarden/desktop-napi";
+import { LogService } from "@bitwarden/logging";
 
-import { DesktopAutotypeService } from "../services/desktop-autotype.service";
+import { WindowMain } from "../../main/window.main";
 
 export class MainDesktopAutotypeService {
-  constructor(private desktopAutotypeService: DesktopAutotypeService) {}
+  keySequence: string = "Alt+CommandOrControl+I";
+
+  constructor(
+    private logService: LogService,
+    private windowMain: WindowMain,
+  ) {}
 
   init() {
-    this.desktopAutotypeService.autotypeEnabled$.subscribe((enabled) => {
-      if (enabled) {
+    ipcMain.on("autofill.configureAutotype", (event, data) => {
+      if (data.enabled === true && !globalShortcut.isRegistered(this.keySequence)) {
         this.enableAutotype();
-      } else {
+      } else if (data.enabled === false && globalShortcut.isRegistered(this.keySequence)) {
         this.disableAutotype();
       }
     });
+
+    ipcMain.on("autofill.completeAutotypeRequest", (event, data) => {
+      const { response } = data;
+
+      if (
+        response.username != undefined &&
+        response.username != null &&
+        response.username.length > 0 &&
+        response.password != undefined &&
+        response.password != null &&
+        response.password.length > 0
+      ) {
+        this.doAutotype(response.username, response.password);
+      }
+    });
+  }
+
+  disableAutotype() {
+    if (globalShortcut.isRegistered(this.keySequence)) {
+      globalShortcut.unregister(this.keySequence);
+    }
+
+    this.logService.info("Autotype disabled.");
   }
 
-  // TODO: this will call into desktop native code
   private enableAutotype() {
-    // eslint-disable-next-line no-console
-    console.log("Enabling Autotype...");
+    const result = globalShortcut.register(this.keySequence, () => {
+      const windowTitle = autotype.getForegroundWindowTitle();
+
+      this.windowMain.win.webContents.send("autofill.listenAutotypeRequest", {
+        windowTitle,
+      });
+    });
 
-    const result = autotype.getForegroundWindowTitle();
-    // eslint-disable-next-line no-console
-    console.log("Window Title: " + result);
+    result
+      ? this.logService.info("Autotype enabled.")
+      : this.logService.info("Enabling autotype failed.");
   }
 
-  // TODO: this will call into desktop native code
-  private disableAutotype() {
-    // eslint-disable-next-line no-console
-    console.log("Disabling Autotype...");
+  private doAutotype(username: string, password: string) {
+    const inputPattern = username + "\t" + password;
+    const inputArray = new Array<number>();
+
+    for (let i = 0; i < inputPattern.length; i++) {
+      inputArray.push(inputPattern.charCodeAt(i));
+    }
```

Comment:
⛏️  (nit): it looks like the `inputArray` would always be of length same as inputPattern, you could preallocate with:

```suggestion
    const inputArray = new Array<number>(inputPattern.length);

    for (let i = 0; i < inputPattern.length; i++) {
      inputArray[i] = inputPattern.charCodeAt(i));
    }
```